File: app/src/main/java/ru/melod1n/vk/mvp/presenter/ConversationsPresenter.java
package ru.melod1n.vk.mvp.presenter;

import androidx.annotation.NonNull;

import java.util.ArrayList;

import ru.melod1n.library.mvp.base.MvpFields;
import ru.melod1n.library.mvp.base.MvpPresenter;
import ru.melod1n.vk.api.model.VKConversation;
import ru.melod1n.vk.mvp.repository.ConversationsRepository;
import ru.melod1n.vk.mvp.view.ConversationsView;

public class ConversationsPresenter extends MvpPresenter<VKConversation, ConversationsView> {

    public ConversationsPresenter(@NonNull ConversationsView view) {
        super(view);

        initRepository(new ConversationsRepository());
    }

    @Override
    protected void insertValues(@NonNull MvpFields fields, @NonNull ArrayList<VKConversation> values) {
        if (view != null) {
            view.insertValues(fields, values);
        }
    }
}

File: app/src/main/java/ru/melod1n/vk/mvp/presenter/ConversationsPresenter.kt
package ru.melod1n.vk.mvp.presenter

import ru.melod1n.vk.api.model.VKConversation
import ru.melod1n.vk.mvp.contract.BaseContract
import ru.melod1n.vk.mvp.repository.ConversationsRepository

class ConversationsPresenter(view: BaseContract.View<VKConversation>) : BaseContract.Presenter<VKConversation>(view) {

    override val tag = "ConversationsPresenter"

    override val repository = ConversationsRepository()

    override var loadedValues = ArrayList<VKConversation>()
    override var cachedValues = ArrayList<VKConversation>()
}